### PR TITLE
support for DMXOPL

### LINKED ISSUE #1987: Add support for DMXOPL GENMIDI lump
Save this for after the next release. Implementation is TBD, but as an example, `OPL3 Emulation` could be split into two "devices":

```
OPL3 Emulation: Vanilla
OPL3 Emulation: DMXOPL
```

Info:
- https://github.com/sneakernets/DMXOPL
- https://doomwiki.org/wiki/DMXOPL

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1987 

dmxopl.op2 was taken from https://github.com/sneakernets/DMXOPL/releases/tag/v2.11d

### CODE REVIEW

File: base/CMakeLists.txt [hunk 1]
```diff
@@ -68,6 +68,7 @@ set(BASE_SOURCES
     all-all/dig090.png
     all-all/dig091.png
     all-all/dig093.png
+    all-all/dmxopl.op2
```
Comment (fabiangreffrath) [inline base/CMakeLists.txt:71]: License?

Comment (rfomin) [reply base/CMakeLists.txt:71]: Added to readme: https://github.com/fabiangreffrath/woof/pull/1988/commits/32e918a7b47d8c75f1c57155e54df49eb9022087

File: src/i_oplmusic.c [hunk 2]
```diff
@@ -1685,7 +1685,8 @@ static const char **I_OPL_DeviceList(void)
     {
         return devices;
     }
-    array_push(devices, "OPL3 Emulation");
+    array_push(devices, "OPL3 Emulation (Vanilla)");
+    array_push(devices, "OPL3 Emulation (DMXOPL)");
```
Comment (ceski-1) [inline src/i_oplmusic.c:1689]: If a separate wad is loaded with a GENMIDI lump, should it be handled like SNDFONT and override the OPL choices? Or should it be appended? Or should it replace "Vanilla" (in which case "Default" is a more appropriate name)?

Let's use `:` to match FluidSynth naming:

```c
    array_push(devices, "OPL3 Emulation: Hello");
    array_push(devices, "OPL3 Emulation: World");
```

Comment (fabiangreffrath) [reply src/i_oplmusic.c:1689]: > Or should it replace "Vanilla" (in which case "Default" is a more appropriate name)?

I guess that's the most sensible thing to do.

If we already call the second choice by the lump name, why not `OPL3 Emulation: GENMIDI` and `OPL3 Emulation: DMXOPL`?

**Comment by fabiangreffrath** [comment]
> dmxopl.op2 was taken from https://github.com/sneakernets/DMXOPL/releases/tag/v2.11d

Why v2.11d, btw and not one of the later releases?

File: src/i_oplmusic.c (hunk 2 above)
Comment (rfomin) [reply src/i_oplmusic.c]: > If a separate wad is loaded with a GENMIDI lump, should it be handled like SNDFONT and override the OPL choices?

Yes, GENMIDI is just a normal lump and would be overridden. This PR adds the DMXOPL lump. We can make "fmbanks" folder, but banks I saw is in wopl format.

> If we already call the second choice by the lump name, why not OPL3 Emulation: GENMIDI and OPL3 Emulation: DMXOPL?

Done in 8081f10

**Comment by rfomin** [comment]
> Why v2.11d, btw and not one of the later releases?

They use WOPL format https://github.com/Wohlstand/libADLMIDI

**Review by ceski-1 — APPROVED** [review]
Works nicely, thanks.

Link to the original discussion for future reference: https://github.com/fabiangreffrath/woof/pull/1907

**Pull request merged**